src/datasets/utils/py_utils.py

```diff
@@ -167,7 +167,8 @@ def _asdict_inner(obj):
             result = {}
             for f in fields(obj):
                 value = _asdict_inner(getattr(obj, f.name))
-                result[f.name] = value
+                if not f.init or value != f.default or f.metadata.get("include_in_asdict_even_if_is_default", False):
+                    result[f.name] = value
```

Comment:
To simplify the JSON and YAML dumps, I stop dumping the dataclass attributes that have are the default value.

e.g. decode=True for Image, or length=-1 for Sequence